core/src/com/unciv/models/ruleset/construction/RejectionReason.kt

```diff
@@ -0,0 +1,26 @@
+package com.unciv.models.ruleset.construction
+
+class RejectionReason(
+    val type: RejectionReasonType,
+    val errorMessage: String,
+    val shouldShow: Boolean
+) {
+    //todo so far the "Used" documentation below is a comprehensive inventory - map to code simplification and proper Kdoc
+
+    // Used in Battle.addXp - true means it's _not_ blocking a unit from being awarded for Great General points
+    // Used in UnitUpgradeManager.canUpgrade - true means the `ignoreRequirements` parameter will filter these away
+    fun techPolicyEraWonderRequirements(): Boolean = type.isTechPolicyEraWonderRequirement
+
+    // Used in CityConstructions.validateInProgressConstructions - true means remove+refund or replace with upgrade + transfer points
+    fun hasAReasonToBeRemovedFromQueue(): Boolean = type.isReasonToDefinitivelyRemoveFromQueue
+
+    // Used in CityConstructionsTable.getConstructionButtonDTOs to filter reasons from being displayed on the button
+    fun isImportantRejection(): Boolean = type.rejectionPrecedence >= 0
+
+    // Used in Battle.addXp - true means it's _not_ blocking a unit from being awarded for Great General points
```

Comment:
Comment is somewhat inaccurate. It is also intended to be used in #10947. Comment left in RejectionReasonType is more accurate: these are intended to be reasons specifically relevant to construction and are to be ignored in cases where units are created outside of the build tab. Useful for quickly checking both whether the unit is replaced and whether you have the correct tech in masse without adding a whole extra set of logic. Note that `.map{ civ.getEquivalentUnit(it) }` is not sufficient, you actually have to do something like `.filter{ civ.getEquivalentUnit(it) == it }` to cover this same use case here which is more expensive than the check that `getRejectionReasons` uses, likely less expensive than checking the entirety of rejection reasons, but the entirety of rejection reasons also gives a free max unit built check, so it's kinda a wash